tests/bluetooth/ctrl_isoal/src/main.c

```diff
@@ -20,1517 +20,7747 @@
+					isoal_global.sink_state[sink_hdl[i]].
+						session.latency_unframed, latency);
+			}
+
+			zassert_equal(res, ISOAL_STATUS_OK,
+				"Sink %d in role %s creation failed!",
+				i, ROLE_TO_STR(role));
+
+			framed = !framed;
+			burst_number++;
+			flush_timeout = (flush_timeout % 3) + 1;
+			sdu_interval_int++;
+			iso_interval_int = iso_interval_int * sdu_interval_int;
+			sdu_interval = (sdu_interval_int * CONN_INT_UNIT_US) - (framed ? 100 : 0);
+			iso_interval = iso_interval_int * CONN_INT_UNIT_US;
+			stream_sync_delay = iso_interval - (200 * i);
+			group_sync_delay = iso_interval - 50;
+		}
+
+		/* Destroy in order */
+		for (int i = 0; i < CONFIG_BT_CTLR_ISOAL_SINKS; i++) {
+			isoal_sink_destroy(sink_hdl[i]);
+
+			zassert_equal(isoal_global.sink_allocated[sink_hdl[i]],
+				ISOAL_ALLOC_STATE_FREE, "Sink destruction failed!");
+
+			zassert_equal(isoal_global.sink_state[sink_hdl[i]].sdu_production.mode,
+				ISOAL_PRODUCTION_MODE_DISABLED, "Sink disable failed!");
+		}
+	}
 }
 
 /**
- * @brief   Trigger payload number sequence error path, with pdu erros
+ * Test Suite	:	RX basic test
+ *
+ * Test error return on exceeding the maximum number of sinks available.
  */
-void test_unframed_seq_pdu_err(void)
+ZTEST(test_rx_basics, test_sink_create_err)
 {
-	clear_sdu_buf();
-
-	construct_pdu_unframed(PDU_BIS_LLID_START_CONTINUE, &testdata[0], 3,
-			       payload_number++, 10000, 0);
-	/* Test recombine */
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	zassert_equal(sink->sdu_production.sdu_written, 3, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_VALID, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
-
-	construct_pdu_unframed(PDU_BIS_LLID_COMPLETE_END, &testdata[3], 7, 123123, 20000, 1);
-	/* Test recombine, should now trigger emit since this is last PDU in SDU */
-	pdu_ref = &testdata[0];
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	/* Expecting no new data to be written but with error status */
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	zassert_equal(sink->sdu_production.sdu_written, 3, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_ERRORS, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
-}
+	isoal_sink_handle_t sink_hdl[CONFIG_BT_CTLR_ISOAL_SINKS+1];
+	uint32_t stream_sync_delay;
+	uint32_t group_sync_delay;
+	uint8_t iso_interval_int;
+	uint8_t flush_timeout;
+	uint32_t sdu_interval;
+	uint8_t burst_number;
+	isoal_status_t res;
+	uint16_t handle;
+	uint8_t role;
+	bool framed;
+
+	handle = 0x8000;
+	role = BT_CONN_ROLE_PERIPHERAL;
+	burst_number = 1;
+	flush_timeout = 1;
+	framed = false;
+	iso_interval_int = 1;
+	sdu_interval = CONN_INT_UNIT_US;
+	stream_sync_delay = CONN_INT_UNIT_US - 200;
+	group_sync_delay = CONN_INT_UNIT_US - 50;
```

Comment:
I feel like there is a lot of values that are seemingly randomly chosen in these tests (and may of them are being used multiple times). 

How are the values chosen? Are some of the values something we can either document and/or reuse for the other tests in this file?

Reply:
The values are somewhat random but there are some relationships.

- ISO Interval > group_sync_delay >= stream_sync_delay
- flush_timeout only matters in the latency calculation but not in the recombination so it's usually fixed at 1
- burst_number should vary depending on the number of PDUs that constitute the SDU or the number that could be received in an ISO interval.
- Relationship between SDU interval and ISO interval depends on the type of PDU (framed / unframed)